src/uwtools/config/templater.py

```diff
@@ -0,0 +1,111 @@
+"""
+Support for rendering Jinja2 templates.
+"""
+
+import logging
+import os
+from typing import Dict, Optional
+
+import uwtools.config.core
+from uwtools.config.j2template import J2Template
+from uwtools.types import DefinitePath, OptionalPath
+from uwtools.utils.file import get_file_type, readable, writable
+
+
+def render(
+    input_file: OptionalPath,
+    output_file: OptionalPath,
+    config_file: DefinitePath,
+    overrides: Optional[Dict[str, str]] = None,
+    values_needed: bool = False,
+    dry_run: bool = False,
+) -> bool:
+    """
+    Render a Jinja2 template.
+
+    :param input_file: Path to the Jinja2 template file to render.
+    :param output_file: Path to the file to write the rendered Jinja2 template to.
+    :param config_file: Path to the config file supplying values to render the template.
+    :param keq_eq_val_pairs: "key=value" strings to supplement config-file values.
+    :param values_needed: Just issue a report about variables needed to render the template?
+    :param dry_run: Run in dry-run mode?
+    """
+    _report(locals())
+    cfgobj = _set_up_config_obj(config_file=config_file, overrides=overrides)
+    with readable(input_file) as f:
+        template_str = f.read()
+    template = J2Template(configure_obj=cfgobj, template_str=template_str)
+    undeclared_variables = template.undeclared_variables
+
+    # If a report of variables required to render the template was requested, make that report and
+    # then return.
+
+    if values_needed:
+        logging.info("Value(s) needed to render this template are:")
+        for var in sorted(undeclared_variables):
+            logging.info(var)
+        return True
```

Comment:
Perhaps a follow on ticket would be helpful given this file ended up getting "created anew" in this PR?

Reply:
I have an idea about how to recover from the `mv` instead of `git mv` situation and will try that out. But I'll also create what should probably be a 1- or 2-point ticket for this refactoring.